src/FFmpegReader.cpp

```diff
@@ -831,8 +837,11 @@ void FFmpegReader::UpdateVideoInfo() {
 		// Yes, a duration was found
 		is_duration_known = true;
 
-		// Calculate number of frames
-		info.video_length = round(info.duration * info.fps.ToDouble());
+		// Calculate number of frames (if not already found in metadata)
+		// Only set this 1 time (this method can be called multiple times)
+		if (info.video_length <= 0) {
+			info.video_length = round(info.duration * info.fps.ToDouble());
```

Comment:
So we only calculate `video_length` from `duration` if we have `duration` and we're missing a value for `video_length`. Because we prefer the data from `nb_frames`? Makes sense.